docs/19_security_audit_document.md

```diff
@@ -0,0 +1,275 @@
+
+The tests also cover invalid data input cases if the rule requires additional data validation.
+
+### Key protection
+
+Metrics Web Application uses [`Firebase Key Restrictions`](https://github.com/platform-platform/monorepo/blob/master/docs/09_firebase_deployment.md#api-key-restrictions) to restrict the services available using the Metrics Firebase project key (also known as `Browser Key`) and restricts the origins this key can be used from.
+
+Currently, the 3 APIs are enabled for the `browser key`:
+- `Identity Toolkit API`.
+- `Token Service API`.
+- `Firebase Installations API`.
+
+### Email domains validation
+
+The application provides additional data protection by allowing to control which email domains are allowed to sign in with Google.
+Application validates the user's email while signing in using the `Google` sign in option. If the provided email does not have the allowed domain, the application denies the sign in attempt.
+
+# CI Integrations
+
+A CI Integrations tool is a command-line application that helps to import build data to the Metrics project making it available in the Metrics Web Application.
+Consider the [following document to learn more about the CI Integrations tool](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/02_ci_integration_user_guide.md).
+
+## Authorization
+
+To be able to import the build data, the CI Integrations tool needs to authorize itself in the `source` CI the build data comes from, and in the `destination` Firebase Cloud Firestore Database.
+
+To authorize in the Firebase, the CI Integration tool uses the `email` and `password` credentials.
+
+CI Integrations tool uses different types of authorization in CIs and uses HTTP headers to pass the authorization credentials. Consider the following table that describes the authorization methods in different CIs used by the CI Integrations tool:
+
+| CI name        | HTTP header     | Value                                      | Authorization type   |
+|----------------|-----------------|--------------------------------------------|----------------------|
+|`Github Actions`| `Authorization` | `Bearer <your_token>`                      | Bearer authorization |
+|`Buildkite`     | `Authorization` | `Bearer <your_token>`                      | Bearer authorization |
+|`Jenkins`       | `Authorization` | `base64(<your_username>:<your_api_token>)` | Basic authorization  |
+
+The CI Integration tool does not store any access credentials and takes them directly from the CI Integrations Config. 
+
+## Configuration
+
+As mentioned, the CI Integrations tool does not store any configuration files or any authorization credentials and parses all the relevant information from the given configuration file.
+
+## Data stored
+
+The CI Integrations tool processes and transfers build data to the Cloud Firestore Database and does not store any data.
+
+# Coverage converter
```

Comment:
For the coverage converter, we should notice that it definitely just converts data - no saving to databases or sending to third-parties, however, the result of convert should be stored within the CI artifacts.